### PR TITLE
Class and import related parsing fixes

### LINKED ISSUE #354: Wrong handling of private Identifierers
Try this:

        class aa {
            #aa = true;
        }

We do not create an instance of PrivateIdentifier. We create a normal node with "#" in the value

### PR DESCRIPTION
**Pull request by jogibear9988**
fixes #354

### CODE REVIEW

File: src/Esprima/JavascriptParser.cs [hunk 1]
```diff
@@ -2074,6 +2078,10 @@ private UnaryExpression ParseBasicUnaryExpression()
         {
             TolerateError(Messages.StrictDelete);
         }
+        if (_context.Strict && unaryExpr.Operator == UnaryOperator.Delete && unaryExpr.Argument is MemberExpression m && m.ChildNodes.Last() is PrivateIdentifier)
```
Comment (adams85) [inline src/Esprima/JavascriptParser.cs:2081]: Enumeration (`m.ChildNodes.Last()`) looks unnecessary here. Wouldn't `m.Property` be enough?

Comment (jogibear9988) [reply src/Esprima/JavascriptParser.cs:2081]: think so

**Comment by lahma** [comment]
Looking really good so far, fixes a lot of parsing problems 👍🏻 Let us know when you think this is mergeable (needs a rebase at least).

**Comment by jogibear9988** [comment]
im done

**Review by lahma — APPROVED** [review]

File: src/Esprima/Scanner.cs [hunk 2]
```diff
@@ -2361,6 +2361,17 @@ internal string EscapeFailingRegex(string pattern)
             pattern = pattern.Replace("[^]", @"[\s\S]");
         }
 
+        if (pattern.Contains("--"))
+        {
+            pattern = pattern.Replace("--[", "-[");
+            var r = new Regex("--([^\\[\\]]*)");
```
Comment (sebastienros) [inline src/Esprima/Scanner.cs:2367]: This should be a static/compiled regex

**Pull request merged**